client/reader/recent/index.tsx

```diff
@@ -1,61 +1,96 @@
-import { Button } from '@wordpress/components';
-import { DataViews, SupportedLayouts, View } from '@wordpress/dataviews';
+import { DataViews, filterSortAndPaginate, SupportedLayouts, View } from '@wordpress/dataviews';
 import { translate } from 'i18n-calypso';
-import { useState, useEffect, useCallback } from 'react';
+import { useState, useEffect, useCallback, useMemo } from 'react';
 import { useSelector, shallowEqual, useDispatch } from 'react-redux';
 import { AnyAction } from 'redux';
 import { ThunkDispatch } from 'redux-thunk';
 import { FullPostView } from 'calypso/blocks/reader-full-post';
+import FormattedHeader from 'calypso/components/formatted-header';
 import { getPostByKey } from 'calypso/state/reader/posts/selectors';
 import { requestPage } from 'calypso/state/reader/streams/actions';
 import { viewStream } from 'calypso/state/reader-ui/actions';
+import RecentPostField from './recent-post-field';
+import RecentSeenField from './recent-seen-field';
+import type { PostItem, ReaderPost } from './types';
 import type { AppState } from 'calypso/types';
 import './style.scss';
 
-interface ReaderPost {
-	site_name: string;
-	postId: number;
-	blogId: number;
-}
-
 const Recent = () => {
 	const dispatch = useDispatch< ThunkDispatch< AppState, void, AnyAction > >();
-
 	const [ selectedItem, setSelectedItem ] = useState< ReaderPost | null >( null );
 
-	const [ view, setView ] = useState( {
-		type: 'list',
-		fields: [ 'post' ],
+	const [ view, setView ] = useState< View >( {
+		type: 'table',
+		fields: [ 'seen', 'post' ],
```

Comment:
I went back and forth on what should be a field. This should allow us to add sorting for unread and post title. Are there other fields that should be sortable?

Reply:
This might be accounted for but publish date is one that comes to mind. Maybe by site name? If I have "all" selected but want the posts "grouped" by each site.

Reply:
Those are great ideas. Publish date might be tricky without making it visible. 🤔 I'll start a thread about it when adding sorting.